cmd/influxd/run/config.go

```diff
@@ -106,3 +116,104 @@ func (c *Config) Validate() error {
 	}
 	return nil
 }
+
+func (c *Config) ApplyEnvOverrides() error {
+	return c.applyEnvOverrides("", reflect.ValueOf(c))
+}
+
+func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value) error {
```

Comment:
Ok, I didn't notice this was a recursive function.

+1 for using `INFLUXDB` (or something shorter) on L121. It'll avoid accidental collisions (particularly for `DATA_DIR` and the plugins) and make it clear which environment variables are being used by `influxd`.